t/10-jobs.t

```diff
@@ -450,7 +450,10 @@ subtest 'carry over, including soft-fails' => sub {
         is($job->result, OpenQA::Jobs::Constants::FAILED, 'job result is failed');
         ok(my $inv = $job->investigate, 'job can provide investigation details');
         ok($inv,                        'job provides failure investigation');
-        is($inv->{last_good}, 99998, 'previous job identified as last good');
+        is(ref(my $last_good = $inv->{last_good}), 'HASH', 'previous job identified as last good and it is a hash');
+        is($last_good->{text}, 99998, 'last_good hash has the text');
+        is($last_good->{type}, 'link', 'last_good hash has the type');
+        is($last_good->{link}, '/test/99998', 'last_good hash has the correct link');
```

Comment:
How about `is_deeply` for the whole thing?

Reply:
I consider this out-of-scope and I don't see the reason to be that explicit. Makes the test just hard to maintain

Reply:
The test is already explicit. I just suggested to merge the checks.

Reply:
Yes but it would make the test fail as soon as any additional key is added – which shouldn't be prevented

Reply:
On the other hand: When adding a new key here, wasn't it nice if that test would fail? Then one would immediately know where to extend the test suite and avoid adding unnecessary test code somewhere else.

But likely it doesn't really matter here.